readablepdf

```diff
@@ -354,12 +358,21 @@ while true; do
 	esac
 done
 
-# Use GNU Parallel to speed things up if it exists.
-if command -v parallel >/dev/null; then
-	parallel tesseract "{}" "{.}" "${TESSERACT_PARAMS}" hocr ::: *.tif
-else
-	for i in ./*.tif; do tesseract "$i" "$(basename "$i")" "${TESSERACT_PARAMS}" hocr; done;
-fi
+
+echo "Estimating CPU-Threads"
+# Estimate CPU Cores (see pdfsandwich code for reference)
+CPU_THREADS=$(cat /proc/cpuinfo | grep processor | awk '{a++} END {print a}')
+export OMP_THREAD_LIMIT=$CPU_THREADS
+
+echo "Using up to $CPU_THREADS threads."
+echo "Running tesseract-ocr"
+
+# if command -v parallel >/dev/null; then
+#	parallel tesseract "{}" "{.}" "${TESSERACT_PARAMS}" hocr ::: *.tif
+# else
+for i in ./*.tif; do tesseract $i $(basename "$i") ${TESSERACT_PARAMS} hocr; done;
```

Comment:
It would seem the opposite is desirable, namely setting  OMP_THREAD_LIMIT=1 and using parallel. (Which doesn't invalidate this as the else condition much the way i wrote it.)

https://twitter.com/ghalfacree/status/1310535190121340929

Reply:
I tried setting OMP_THREAD_LIMIT=1 and use 16 workers, but it was slow as hell. I therefore tried a combination of tesseracts own multithreading and parallels, so multiple threads can work on a single image. That sped things up, but maybe I did something the wrong way. Maybe you could test it again on your machine? (see later commits for the multithreaded version)